src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1086,6 +1146,9 @@ public bool LoadCurrentSegment()
             public void UpdateBaseline(long timestampDiff)
             {
                 Debug.Assert(timestampDiff > 0);
+                if (timestampDiff == 0)
+                    TryMergeIncrementResults(_context, this);
```

Comment:
What happens if this returns false?

Reply:
In this scenario, we reach 2KB with equal timestamps and different tag values. I can throw an exception if this is false or return immediately from this point.  but I think we should handle it better (maybe remove all elements left in append/increment operations with this timestamp?)